### ISSUE TITLE
Conditional dependencies works only when specified in requirements.yaml and not in charts.yaml

### ISSUE THREAD

**Issue by pjastrzabek** [helm/helm#8417]
**Output of `helm version`:**
version.BuildInfo{Version:"v3.2.4", GitCommit:"0ad800ef43d3b826f31a5ad8dfbb4fe05d143688", GitTreeState:"dirty", GoVersion:"go1.14.3"}

**Output of `kubectl version`:**
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.4", GitCommit:"c96aede7b5205121079932896c4ad89bb93260af", GitTreeState:"clean", BuildDate:"2020-06-18T02:57:24Z", GoVersion:"go1.14.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:43:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

minikube

I tried to use conditional helm dependency and it seems that it works only if dependency is specified in requirements.yaml file.
If I move it to charts.yaml it's ignored (so dependency is always installed)

A lot of documentation not directly, but suggests that dependencies can be specified in chart.yaml file
For example here: https://helm.sh/docs/helm/helm_dependency/

But conditional expression seems to not work in such case.

Is it a bug and if so. Is it in helm or rather documentation could be improved?

**Comment by pjastrzabek** [comment]
To be more specific.

This is my values.yaml

mysql:
  nameOverride: db
  root:
    password: xxx
  enabled: false

and requirements.yaml

dependencies:
  - name: mysql
    condition: mysql.enabled
    alias: helmmysql
    version: ~6.14.4
    repository: https://charts.bitnami.com/bitnami

In such setup all works (dependency is not installed as value of mysql.enabled is false)

If I move dependencies section to charts.yaml dependency is installed **always**.

**Comment by jeff-knurek** [comment]
@pjastrzabek I wasn't able to reproduce. I used the same values that you provided, but maybe the problem is a yaml formatting issue _(it's hard to tell without the formatting in your github post)_.

I have a `Chart.yaml` with the following:
```yaml
apiVersion: v2
name: my-test
description: A Helm chart for Kubernetes
type: application
version: 0.1.0
appVersion: 1.16.0
dependencies:
  - name: mysql
    condition: mysql.enabled
    alias: helmmysql
    version: ~6.14.4
    repository: https://charts.bitnami.com/bitnami
```
and the `values.yaml` file with this, where i turn on/off the mysql.enabled 
```yaml
replicaCount: 1

image:
  repository: nginx
  pullPolicy: IfNotPresent
  tag: ""

mysql:
  enabled: false
  nameOverride: db
  root:
    password: xxx
```

and I run `helm install example my-test` and in between disabling/enabling the dependency I delete the release: `helm delete example`

**Comment by pjastrzabek** [comment]
I prepared sample project here
https://github.com/pjastrzabek/helm_dependency_bug

overriding from command line works as expected (--set mysql.enabled=false) so for whatever reason values in values.yaml
seems to be ignored.

According to doc. if expression is not found it's being ignored so I guess this is what happens..

My helm version is

version.BuildInfo{Version:"v3.2.4", GitCommit:"0ad800ef43d3b826f31a5ad8dfbb4fe05d143688", GitTreeState:"dirty", GoVersion:"go1.14.3"}

**Comment by jeff-knurek** [comment]
works for me
```console
$ git clone https://github.com/pjastrzabek/helm_dependency_bug.git
$ helm dep update helm_dependency_bug/
$ helm install example helm_dependency_bug
$ kubectl get pods
NAME                                         READY   STATUS    RESTARTS   AGE
example-helmdependencybug-65bc55bfdb-mdgrm   1/1     Running   0          7s
$ helm delete example 
$ kubectl get pods
No resources found in default namespace.
```
`edit the values.yaml file to set mysql.enabled to true`
```console
$ helm install example helm_dependency_bug
$ kubectl get pods
NAME                                         READY   STATUS    RESTARTS   AGE
example-helmdependencybug-65bc55bfdb-d8fhs   0/1     Running   0          5s
example-mysql-master-0                       0/1     Running   0          5s
example-mysql-slave-0                        0/1     Running   0          5s
```
---
looking at the example you gave, I saw that in the Chart.yaml file, you've added 
```yaml
mysql:
  enabled: false
```
this isn't going to have any effect. The value you need to change is in `values.yaml`

**Comment by pjastrzabek** [comment]
As described in readme I started verifying whether dependencies are being used by this command:

helm upgrade --install --debug --dry-run goodly-guppy  -v 3 ./  | less

With a --debug flag mysql services and deployments are being rendered regardless of mysql.enabled value.
When I remove --debug flag they disappear.

When I try to install package as you do, now I cannot reproduce the issue.
I believe the issue existed originally and only after that I switched to using --dry-run and --debug flags.
But now all seems to work (helm 3.3.1)

Thank you @jeff-knurek for reply. Closing the issue.
